webapp/plugins/index.js

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+// EXPERIMENTAL - SUBJECT TO CHANGE
+
+import store from 'stores/redux_store.jsx';
+
+global.window.plugins = {};
+global.window.plugins.components = {};
+
+export function registerComponents(components) {
+    global.window.plugins.components = {...components, ...global.window.plugins.components};
```

Comment:
Not entirely sure if storing these globally is the best solution but I'm not sure of any great alternatives. We _could_ store them in the redux store but I can't see any real benefit to that